sdk/identity/Azure.Identity/src/DeviceCodeCredential.cs

```diff
@@ -224,23 +223,17 @@ private async ValueTask<AccessToken> GetTokenImplAsync(bool async, TokenRequestC
 
         private async Task<AccessToken> GetTokenViaDeviceCodeAsync(string[] scopes, bool async, CancellationToken cancellationToken)
         {
-            AuthenticationResult result = await _client.AcquireTokenWithDeviceCodeAsync(scopes, code => DeviceCodeCallback(code, cancellationToken), async, cancellationToken).ConfigureAwait(false);
+            AuthenticationResult result = await Client.AcquireTokenWithDeviceCodeAsync(scopes, code => DeviceCodeCallbackImpl(code, cancellationToken), async, cancellationToken).ConfigureAwait(false);
 
-            _record = new AuthenticationRecord(result, _clientId);
+            Record = new AuthenticationRecord(result, ClientId);
 
             return new AccessToken(result.AccessToken, result.ExpiresOn);
         }
 
-        private Task DeviceCodeCallback(DeviceCodeResult deviceCode, CancellationToken cancellationToken)
+        private Task DeviceCodeCallbackImpl(DeviceCodeResult deviceCode, CancellationToken cancellationToken)
```

Comment:
Nit - use `Core`.  We're supposed to avoid `Impl` in private names because the stack traces are uglier.